lib/middleman-simple-thumbnailer/extension.rb

```diff
@@ -37,12 +37,35 @@ def after_build(builder)
       def image_tag(path, options={})
         resize_to = options.delete(:resize_to)
         return super(path, options) unless resize_to
+        ext = app.extensions[:middleman_simple_thumbnailer]
 
-        image = MiddlemanSimpleThumbnailer::Image.new(path, resize_to, app)
         if app.development?
-          super("data:#{image.mime_type};base64,#{image.base64_data}", options)
+
+          # For better performance on local development
+          # (high bandwidth)
+          # simply change width in <img>-tag
+          
+          original_width  = options.delete(:width)
+          resized_width   = resize_to.split('x').first.to_i
+          original_height = options.delete(:height)
+          resized_height  = resize_to.split('x').last.to_i
+          
+          if !original_width.nil? && original_width <= resized_width
+            if original_height <= resized_height
+              options[:width] = original_width
```

Comment:
As I tried to tell  you, I am not the original author of this extension.
However I think that the goal was to have a good preview of the result during development.
I have looked to middleman, and the "preview server" is an instance of the "standard" ruby "webrick" server. This http server is quite powerful and allow you to 'mount' servlets like file handler to path. 

Therefore at the extension initialisation, if we could grab the instance of the web server we could map a special path (/__thumbnail_cache for example) to the location of the cache folder.

This would not solve the fact that the images would need to be resized at least once, but it would solve the problem of having heavy data url on the page...